unittests/currency_tests.cpp

```diff
@@ -25,11 +25,8 @@ class currency_tester : public T {
       auto push_action(const account_name& signer, const action_name &name, const variant_object &data ) {
          string action_type_name = abi_ser.get_action_type(name);
 
-         action act;
-         act.account = "eosio.token"_n;
-         act.name = name;
-         act.authorization = vector<permission_level>{{signer, config::active_name}};
-         act.data = abi_ser.variant_to_binary(action_type_name, data, abi_serializer::create_yield_function( T::abi_serializer_max_time ));
+         action act(vector<permission_level>{{signer, config::active_name}}, "eosio.token"_n, name,
+                    abi_ser.variant_to_binary(action_type_name, data, abi_serializer::create_yield_function( T::abi_serializer_max_time )));
 
```

Comment:
I feel the original way of initializing individual fields explicitly clearer and more readable. `action` is a data class and used most locally. It is probably appropriate to be a `struct` without const.

Reply:
I very much disagree. Because `action` has constructors, it implies that the object should be constructed using the provided constructors. 

The direct member initialization forgoes any hope to ensure that objects are always constructed in a consistent fashion, and that some internal members possibly dependent on the constructors parameters are always initialized correctly.

Reply:
https://github.com/AntelopeIO/spring/pull/951/commits/b158e5b2b187b226e59e58eb64cb735a21f8d696

Reply:
@arhag do you have an opinion on whether it is worthwhile to merge this PR?